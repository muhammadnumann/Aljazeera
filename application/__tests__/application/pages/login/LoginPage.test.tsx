import React from 'react'
import Login from '../../../../pages/login'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

beforeAll(() => {
    jest.mock('react-cookie');
    jest.mock('@vidispine/vdt-api');
});

describe('Login Page', () => {

    it('should render correctly', () => {
        const { asFragment } = render(<Login />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should contain credential fields', () => {
        render(<Login />);

        const usernameField = screen.getByTestId('Username');
        expect(usernameField).toBeVisible();
        expect(usernameField).toBeEnabled();

        const passwordField = screen.getByTestId('Password');
        expect(passwordField).toBeVisible();
        expect(passwordField).toBeEnabled();

        const loginButton = screen.getByTestId('Sign in');
        expect(loginButton).toBeVisible();
        expect(loginButton).toBeEnabled();
    });
});
