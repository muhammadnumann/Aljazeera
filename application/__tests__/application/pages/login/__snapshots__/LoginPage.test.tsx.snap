// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center dark:bg-grayDark-200 bg-gray-50 overflow-hidden h-screen w-full relative"
  >
    <div
      class="lg:w-8/12 w-full h-full bg-[url('../public/undefined/library.png')] bg-no-repeat bg-center bg-cover lg:blur-none blur-sm"
    />
    <div
      class="lg:relative lg:block flex justify-center items-center absolute lg:mx-16 lg:w-4/12 w-full lg:p-0 sm:p-5 top-[25%] lg:left-0 lg:top-0"
    >
      <div>
        <div
          class="flex items-center"
        >
          <div
            class="grow-0 shrink-1 lg:shrink-0 basis-auto"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt=""
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="ml-4 pl-0.5"
          >
            <h1
              class="text-3xl text-gray-900 dark:text-grayDark-900 gray-900 font-bold"
            >
              AJ Digital MAM
            </h1>
          </div>
        </div>
        <h2
          class="text-2xl text-gray-900 dark:text-grayDark-900 mt-14 font-bold"
        >
          Sign in to your account
        </h2>
        <div
          class="pb-5 pt-2 mt-5 max-w-xs"
        >
          <label
            class="font-semibold text-gray-900 dark:text-grayDark-900"
          >
            Username
          </label>
          <div
            class="undefined undefined"
          >
            <div>
              <div>
                <div
                  class="relative"
                >
                  <div
                    class="absolute top-4 left-3"
                  >
                    <i
                      class="fa fa-search text-gray-400 z-20 hover:text-gray-400"
                    />
                  </div>
                  <input
                    class="block px-2.5 pb-2.5 pt-3 text-sm border-1 border-gray-300 appearance-none  dark:border-gray-600 focus:ring-0 focus:border-blue-600 peer w-full pl-2 rounded-lg z-0 focus:shadow focus:outline-none dark:text-grayDark-800 text-gray-800 bg-gray-50 dark:bg-gray-700
                border-gray-200
              h-9
              mt-1 border-gray-300 !rounded-full
            "
                    data-testid="Username"
                    placeholder="you@example.com"
                    type="text"
                    value=""
                  />
                  <label
                    class="absolute text-sm text-gray-500 dark:text-gray-400 duration-300 transform -translate-y-4 scale-75 top-2 z-10 origin-[0] bg-white dark:bg-gray-700 px-2 peer-focus:px-2 peer-focus:text-blue-600 dark:peer-focus:text-gray-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:top-1/2 peer-focus:top-2 peer-focus:scale-75 peer-focus:-translate-y-4 left-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pb-2 max-w-xs"
        >
          <label
            class="font-semibold text-gray-900 dark:text-grayDark-900"
          >
            Password
          </label>
          <div
            class="undefined undefined"
          >
            <div>
              <div>
                <div
                  class="relative"
                >
                  <div
                    class="absolute top-4 left-3"
                  >
                    <i
                      class="fa fa-search text-gray-400 z-20 hover:text-gray-400"
                    />
                  </div>
                  <input
                    class="block px-2.5 pb-2.5 pt-3 text-sm border-1 border-gray-300 appearance-none  dark:border-gray-600 focus:ring-0 focus:border-blue-600 peer w-full pl-2 rounded-lg z-0 focus:shadow focus:outline-none dark:text-grayDark-800 text-gray-800 bg-gray-50 dark:bg-gray-700
                border-gray-200
              h-9
              mt-1 p-2 border-gray-300 !rounded-full
            "
                    data-testid="Password"
                    placeholder="*********"
                    type="password"
                    value=""
                  />
                  <label
                    class="absolute text-sm text-gray-500 dark:text-gray-400 duration-300 transform -translate-y-4 scale-75 top-2 z-10 origin-[0] bg-white dark:bg-gray-700 px-2 peer-focus:px-2 peer-focus:text-blue-600 dark:peer-focus:text-gray-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:top-1/2 peer-focus:top-2 peer-focus:scale-75 peer-focus:-translate-y-4 left-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="py-2"
        >
          <button
            class="inline-flex items-center px-4 py-2 border border-transparent rounded-md font-semibold text-xs 
                text-white tracking-widest transition ease-in-out duration-150 
                false w-full justify-center bg-blue-800 active:bg-blue-800 mt-5 font-semibold text-base !border-0 !rounded-full max-w-xs"
            data-testid="Sign in"
            type="submit"
          >
            Sign in
          </button>
        </div>
        <label
          class="dark:text-red-700 text-red-900"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
